### PR TITLE
CD pipeline that deploys resources to GKE cluster

### PR DESCRIPTION
**Pull request by KubaKrzych**
The product of this branch is a completed CD pipeline that deploys resources to GKE. 

The mentioned resources are Docker images stored in GCR and/or images stored in Docker hub.

I'll be grateful for any tips and comments.

### CODE REVIEW

File: k8s/values-backend.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            memory: 300Mi
+            cpu: 200m
+          limits:
+            memory: 400Mi
+            cpu: 200m
+  env:
+    - name: POSTGRES_HOST
+      value: "database-service"
+    - name: POSTGRES_PASSWORD
+      value: "postgres"
```
Comment (matined) [inline k8s/values-backend.yaml:21]: we need to store this value in secrets, I created POSTGRES_PASSWORD

File: k8s/values-backend.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+deployment:
+  replicas: 1
+  name: api
```
Comment (matined) [inline k8s/values-backend.yaml:3]: let's stay consistent and use backend instead of api everywhere

File: k8s/values-database.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            cpu: 200m
+          limits:
+            memory: 300Mi
+            cpu: 200m
+  env:
+    - name: POSTGRES_USER
+      value: "postgres"
+    - name: POSTGRES_PASSWORD
+      value: "postgres"
+    - name: POSTGRES_DB
```
Comment (matined) [inline k8s/values-database.yaml:22]: I created two more secrets: POSTGRES_USER, POSTGRES_DATABASE

File: k8s/values-backend.yaml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  template:
+    spec:
+      container:
+        image: gcr.io/${{ secrets.GOOGLE_PROJECT_ID }}/backend
```
Comment (matined) [inline k8s/values-backend.yaml:7]: I wonder if it works

**Review by matined — APPROVED** [review]

File: k8s/values-backend.yaml (hunk 4 above)
Comment (KubaKrzych) [reply k8s/values-backend.yaml:7]: Well, I'm positive that it doesn't work. Do you think a better option would be to pass a project_id as an environmental variable?

I'm if the GCP project ID is a vulnerable data, if not then I can pass the project ID explicitily.

**Comment by matined** [comment]
One more request is to rename `k8s` folder to `kubernetes`

**Review by matined — APPROVED** [review]

**Comment by KubaKrzych** [comment]
I'm pretty sure I've added and changed everything you've mentioned. 

I've decided to stick with the deployment of backend/front/... in a way it was before. I've left untouched those if-statements that we've chatted about. In the long term I think we should think about something else, but it's cool for now.

**Pull request merged**